### PR TITLE
New api

### PR DESCRIPTION
**Pull request by dneise**
we talked about a different API for the crawler involving namedtuples and units ...

here is a proposal

### CODE REVIEW

File: smart_fact_crawler/__init__.py [hunk 1]
```diff
@@ -109,36 +42,36 @@ def drive():
...
-    return {
-        'Time_Stamp': smartfact_time2datetime(tc[0, 0]),
-        'Magnitude': str2float(tc[1, 1]),
-        'Sensor_Frequency_in_Hz': str2float(tc[2, 1]),
-        'Sensor_Period_in_s': str2float(tc[4, 1]),
-        'Sensor_Temperature_in_C': str2float(tc[5, 1]),
-    }
+    return to_namedtuple('SqmPage', {
+        'timestamp': (sft2dt(tc[0, 0]), None),
+        'magnitude': (s2f(tc[1, 1]), '???'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: magnitude is dimensionless (log(lightflux / reference light flux))

File: smart_fact_crawler/__init__.py [hunk 2]
```diff
@@ -109,36 +42,36 @@ def drive():
...
+        'sensor_frequency': (s2f(tc[2, 1]), 'Hz'),
+        'sensor_period': (s2f(tc[4, 1]), 's'),
+        'sensor_temperature': (s2f(tc[5, 1]), 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: 'C' is coulomb. Pint understands 'degC' and 'celsius'.

File: smart_fact_crawler/__init__.py [hunk 3]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
-        'Wind_speed_in_km_per_h': str2float(tc[7, 1]),
-        'Wind_gusts_in_km_per_h': str2float(tc[8, 1]),
-        'Wind_direction': tc[9, 1],
-        'Dust_TNG_in_ug_per_m3': str2float(tc[10, 1]),
-    }
+    return to_namedtuple('WeatherPage', {
+        'timestamp': (sft2dt(tc[0, 0]), None),
+        'sun': (tc[1, 1], '%'),
+        'moon': (tc[2, 1], '%'),
+        'temperature': (s2f(tc[3, 1]), 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 4]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
+        'dew_point': (s2f(tc[4, 1]), 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 5]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
     tc = TableCrawler(url)
-    return {
-        'Time_Stamp': smartfact_time2datetime(tc[0, 0]),
-        '24h_min_temperature_in_C': tc[1, 1],
-        'Current_temperature_in_C': tc[2, 1],
-        '24h_max_temperature_in_C': tc[3, 1],
-    }
+    return to_namedtuple('ContainerTemperaturePage', {
+        'timestamp': (sft2dt(tc[0, 0]), None),
+        'daily_min': (tc[1, 1], 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 6]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
+        'current': (tc[2, 1], 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 7]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
-        'Time_Stamp': smartfact_time2datetime(tc[0, 0]),
-        'Avg_humidity_in_percent': str2float(tc[1, 1]),
-        'Max_rel_temp_in_C': str2float(tc[2, 1]),
-        'Avg_rel_temp_in_C': str2float(tc[3, 1]),
-        'Min_rel_temp_in_C': str2float(tc[4, 1]),
-    }
+    return to_namedtuple('CameraClimatePage', {
+        'timestamp': (sft2dt(tc[0, 0]), None),
+        'humidity_mean': (s2f(tc[1, 1]), '%'),
+        'relative_temperature_max': (s2f(tc[2, 1]), 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 8]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
+        'relative_temperature_mean': (s2f(tc[3, 1]), 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 9]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
+        'relative_temperature_min': (s2f(tc[4, 1]), 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 10]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
-        'System_Status': ' '.join(tc[1, 1:]),
-        'Rel_camera_temp_in_C': str2float(tc[3, 1]),
-        'Humidity_in_Percent': str2float(tc[4, 1]),
-        'Wind_speed_in_km_per_h': str2float(tc[4, 2]),
-    }
+    return to_namedtuple('MainPage', {
+        'timestamp_1': (sft2dt(tc[0, 0]), None),
+        'timestamp_2': (sft2dt(tc[0, 1]), None),
+        'system_status': (' '.join(tc[1, 1:]), None),
+        'relative_camera_temperature': (s2f(tc[3, 1]), 'C'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: same as above

File: smart_fact_crawler/__init__.py [hunk 11]
```diff
@@ -148,166 +81,159 @@ def next_datetime_from_hhmm_string(hhmm):
...
 
 def trigger_rate(url=smartfacturl + 'trigger.data'):
     tc = TableCrawler(url)
-    return {
-        'Time_Stamp': smartfact_time2datetime(tc[0, 0]),
-        'Trigger_Rate_in_1_per_s': str2float(tc[1, 1]),
-    }
+    return to_namedtuple('TriggerPage', {
+        'timestamp': (sft2dt(tc[0, 0]), None),
+        'trigger_rate': (s2f(tc[1, 1]), '1/s'),
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: yay!

**Comment by dneise** [comment]
I entirely agree with Max, but I'm offline for 8 days now... so if the new API is needed soon maybe someone else needs to deal with these issues.

**Comment by MaxNoe** [comment]
Any Preferences? `degC` or `celsius`?

**Comment by dneise** [comment]
I'd say degC

**Comment by MaxNoe** [comment]
Aaaarg.

`astropy.units.Unit` and `pint.UnitRegistry` have no common alias for `degrees celsius`

Astropy supports `Celsius` and `deg_C`, Pint supports `celsius` and `degC`.

I would go for `deg_C` because `'deg_C'.replace('_', '')`  is easy to do has no side effects.

**Comment by MaxNoe** [comment]
It's even easier:

```{python}
from pint import UnitRegistry
import smart_fact_crawler as sfc

u = UnitRegistry()
u.define('deg_C = degC')

u.Quantity(*sfc.camera_climate().relative_temperature_max)
```

**Comment by MaxNoe** [comment]
This would also fix #4

**Comment by relleums** [comment]
Very nice. Now that I see it with units I like it a lot. I also like the 'deg_C' solution for the temperature unit. You guys also fixed my wild west key names upper and lower case mix. I am for merging back into the master.

**Pull request merged**